src/rgw/rgw_common.cc

```diff
@@ -987,7 +987,7 @@ bool verify_object_permission(struct req_state * const s,
   }
 
   bool ret = object_acl->verify_permission(*s->auth_identity, s->perm_mask, perm);
-  if (ret) {
+  if (ret && s->cct->_conf->rgw_swift_verify_object_acl) {
```

Comment:
thanks! we may need to consider the `rgw_enforce_swift_acls`. how about this change @cbodley ?
```cpp
bool verify_object_acl = !(s->prot_flags & RGW_REST_SWIFT) ||
                            s->cct->_conf->rgw_swift_verify_object_acl ||
                            !s->cct->_conf->rgw_enforce_swift_acls;

if (verify_object_acl) {
  ret = object_acl->verify_permission(*s->auth_identity, s->perm_mask, perm);
  if (ret) {
    return true;  
  }
}
```